apps/web/pages/api/v1/environments/[environmentId]/surveys/[surveyId]/duplicate/[targetEnvironmentId].ts

```diff
@@ -0,0 +1,159 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+export default async function handle(req: NextApiRequest, res: NextApiResponse) {
+  const environmentId = req.query.environmentId?.toString();
+  const targetEnvironmentId = req.query.targetEnvironmentId?.toString();
+
+  const surveyId = req.query.surveyId?.toString();
+
+  if (environmentId === undefined) {
+    return res.status(400).json({ message: "Missing environmentId" });
+  }
+  if (surveyId === undefined) {
+    return res.status(400).json({ message: "Missing surveyId" });
+  }
+
+  const hasAccess = await hasEnvironmentAccess(req, res, environmentId);
+  const hasTargetEnvAccess = await hasEnvironmentAccess(req, res, targetEnvironmentId);
+
+  if (!hasAccess || !hasTargetEnvAccess) {
+    return res.status(403).json({ message: "Not authorized" });
+  }
+
+  // POST
+  else if (req.method === "POST") {
+    // duplicate current survey including its triggers
+    const existingSurvey = await prisma.survey.findFirst({
+      where: {
+        id: surveyId,
+        environmentId,
+      },
+      include: {
+        triggers: {
+          include: {
+            eventClass: true,
+          },
+        },
+        attributeFilters:{
+          include:{
+            attributeClass:true
+          }
+        }
+      },
+    });
+
+    if (!existingSurvey) {
+      return res.status(404).json({ message: "Survey not found" });
+    }
+
+    let targetEnvironmentTriggers: string[] = [];
+    // map the local triggers to the target environment
+    for (let i = 0; i < existingSurvey.triggers.length; i++) {
+      const trigger = existingSurvey.triggers[i];
+      const targetEnvironmentTrigger = await prisma.eventClass.findFirst({
+        where: {
+          name: trigger.eventClass.name,
+          environment: {
+            id: targetEnvironmentId,
+          },
+        },
+      });
+      if (!targetEnvironmentTrigger) {
+        // if the trigger does not exist in the target environment, create it
+        const newTrigger = await prisma.eventClass.create({
+          data: {
+            name: trigger.eventClass.name,
+            environment: {
+              connect: {
+                id: targetEnvironmentId,
+              },
+            },
+            createdAt: new Date(),
+            updatedAt: new Date(),
+            description: trigger.eventClass.description,
+            type: trigger.eventClass.type,
+            noCodeConfig: trigger.eventClass.noCodeConfig
+              ? JSON.parse(JSON.stringify(trigger.eventClass.noCodeConfig))
+              : undefined,
+          },
+        });
+        targetEnvironmentTriggers.push(newTrigger.id);
+      } else {
+        targetEnvironmentTriggers.push(targetEnvironmentTrigger.id);
+      }
+    }
+
+
+    let targetEnvironmentAttributeFilters: string[] = [];
+    // map the local attributeFilters to the target env
+    for (let i = 0; i < existingSurvey.attributeFilters.length; i++) {
```

Comment:
a `for...of` loop is much easier to use for this functionality 😊
you can just write
```ts
for (const attributeFilter of existingSurvey.attributeFilters) {
...
}
```